packages/component-library/src/components/overlay/mt-modal/mt-modal.vue

```diff
@@ -0,0 +1,205 @@
+import MtModalClose from "./mt-modal-close.vue";
+import { createId } from "@/utils/id";
+import * as focusTrap from "focus-trap";
+
+defineProps<{ title: string }>();
+
+const id = `mt-modal--${createId()}`;
+
+const { isOpen, setIsOpen } = useModalContext("mt-modal");
+
+let closeOnEscapeEventListener: ((event: KeyboardEvent) => void) | undefined = undefined;
+
+watch(isOpen, (value) => {
+  if (value) {
+    closeOnEscapeEventListener = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener("keydown", closeOnEscapeEventListener);
+  } else {
+    if (closeOnEscapeEventListener) {
+      document.removeEventListener("keydown", closeOnEscapeEventListener);
+    }
+  }
+});
+
+onUnmounted(() => {
+  if (closeOnEscapeEventListener) {
+    document.removeEventListener("keydown", closeOnEscapeEventListener);
+  }
+});
+
+const modalRef = ref<HTMLElement | null>(null);
+let trap: ReturnType<typeof focusTrap.createFocusTrap> | undefined;
+
+watch(
+  isOpen,
+  async () => {
+    const isModalOpen = !!isOpen.value;
+    if (isModalOpen) await nextTick();
+
+    if (!modalRef.value) return;
+    if (isModalOpen) {
+      trap = focusTrap.createFocusTrap(modalRef.value, {
+        tabbableOptions: { displayCheck: "none" },
+        allowOutsideClick: true,
+      });
+
+      trap.activate();
+
+      return;
+    }
+
+    if (trap) trap.deactivate();
+  },
+  { immediate: true },
+);
+
+onUnmounted(() => {
+  if (trap) trap.deactivate();
+});
+</script>
+
+<style scoped lang="scss">
+.mt-modal {
+  height: auto;
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  translate: -50% -50%;
+  z-index: 2000;
+  background-color: var(--color-elevation-surface-raised);
+  border-radius: 0.5rem;
+  overflow: hidden;
+  width: min(660px, calc(100vw - 2rem));
+  border: 1px solid var(--color-border-primary-default);
+  max-height: calc(100vh - 2rem);
+  display: flex;
+  flex-direction: column;
+}
+
+.modal-enter-active {
+  transition: opacity 400ms cubic-bezier(0.05, 0.7, 0.1, 1);
+
+  @media (prefers-reduced-motion: no-preference) {
+    transition:
+      opacity 400ms cubic-bezier(0.05, 0.7, 0.1, 1),
+      scale 400ms cubic-bezier(0.05, 0.7, 0.1, 1);
+  }
+}
+
+.modal-leave-active {
+  transition: opacity 200ms cubic-bezier(0.3, 0, 0.8, 0.15);
+
+  @media (prefers-reduced-motion: no-preference) {
+    transition:
+      opacity 200ms cubic-bezier(0.3, 0, 0.8, 0.15),
+      scale 200ms cubic-bezier(0.3, 0, 0.8, 0.15);
+  }
+}
+
+.modal-enter-from,
+.modal-leave-to {
+  opacity: 0;
+
+  @media (prefers-reduced-motion: no-preference) {
+    scale: 0.9;
+  }
+}
+
+.mt-modal__content {
+  --mt-modal-padding: 1.5rem;
+  --mt-modal-padding-bottom: 2.5rem;
+  --inset-top: var(--mt-modal-padding);
+  --inset-right: var(--mt-modal-padding);
+  --inset-bottom: var(--mt-modal-padding-bottom);
+  --inset-left: var(--mt-modal-padding);
+
+  padding: var(--mt-modal-padding);
```

Comment:
```suggestion
  padding: 1.5rem;
```

Let's simplify (and optically fix) this but having all content paddings the same 😊